stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -534,178 +536,192 @@ julia> A = [1 -2 -3; 2 3 -1]
 
 """
-    norm(A::Adjoint{<:Any,<:AbstracVector}, q::Real=2)
-    norm(A::Transpose{<:Any,<:AbstracVector}, q::Real=2)
+    opnorm(A::Adjoint{<:Any,<:AbstracVector}, q::Real=2)
+    opnorm(A::Transpose{<:Any,<:AbstracVector}, q::Real=2)
 
-For Adjoint/Transpose-wrapped vectors, return the ``q``-norm of `A`, which is
-equivalent to the p-norm with value `p = q/(q-1)`. They coincide at `p = q = 2`.
+For Adjoint/Transpose-wrapped vectors, return the operator ``q``-norm of `A`, which is
+equivalent to the `p`-norm with value `p = q/(q-1)`. They coincide at `p = q = 2`.
+Use [`norm`](@ref) to compute the `p` norm of `A` as a vector.
 
 The difference in norm between a vector space and its dual arises to preserve
 the relationship between duality and the inner product, and the result is
-consistent with the p-norm of `1 × n` matrix.
+consistent with the operator `p`-norm of `1 × n` matrix.
 
 # Examples
 ```jldoctest
 julia> v = [1; im];
 
 julia> vc = v';
 
-julia> norm(vc, 1)
+julia> opnorm(vc, 1)
 1.0
 
+julia> norm(vc, 1)
+2.0
+
 julia> norm(v, 1)
 2.0
 
+julia> opnorm(vc, 2)
+1.4142135623730951
+
 julia> norm(vc, 2)
 1.4142135623730951
 
 julia> norm(v, 2)
 1.4142135623730951
 
-julia> norm(vc, Inf)
+julia> opnorm(vc, Inf)
 2.0
 
+julia> norm(vc, Inf)
+1.0
+
 julia> norm(v, Inf)
 1.0
 ```
 """
-norm(v::TransposeAbsVec, q::Real) = q == Inf ? norm(v.parent, 1) : norm(v.parent, q/(q-1))
-norm(v::AdjointAbsVec, q::Real) = q == Inf ? norm(conj(v.parent), 1) : norm(conj(v.parent), q/(q-1))
-norm(v::AdjointAbsVec) = norm(conj(v.parent))
-norm(v::TransposeAbsVec) = norm(v.parent)
+opnorm(v::TransposeAbsVec, q::Real) = q == Inf ? norm(v.parent, 1) : norm(v.parent, q/(q-1))
+opnorm(v::AdjointAbsVec, q::Real) = q == Inf ? norm(conj(v.parent), 1) : norm(conj(v.parent), q/(q-1))
+opnorm(v::AdjointAbsVec) = norm(conj(v.parent))
+opnorm(v::TransposeAbsVec) = norm(v.parent)
 
-function vecdot(x::AbstractArray, y::AbstractArray)
+norm(v::Union{TransposeAbsVec,AdjointAbsVec}, p::Real) = norm(v.parent, p)
+
+function inner(x::AbstractArray, y::AbstractArray)
     lx = _length(x)
     if lx != _length(y)
         throw(DimensionMismatch("first array has length $(lx) which does not match the length of the second, $(_length(y))."))
     end
     if lx == 0
-        return dot(zero(eltype(x)), zero(eltype(y)))
+        return inner(zero(eltype(x)), zero(eltype(y)))
     end
-    s = zero(dot(first(x), first(y)))
+    s = zero(inner(first(x), first(y)))
     for (Ix, Iy) in zip(eachindex(x), eachindex(y))
-        @inbounds  s += dot(x[Ix], y[Iy])
+        @inbounds s += inner(x[Ix], y[Iy])
     end
     s
 end
 
 """
-    vecdot(x, y)
+    inner(x, y)
 
 For any iterable containers `x` and `y` (including arrays of any dimension) of numbers (or
-any element type for which `dot` is defined), compute the Euclidean dot product (the sum of
-`dot(x[i],y[i])`) as if they were vectors.
+any element type for which `inner` is defined), compute the inner product (or dot product
```

Comment:
"Euclidean inner product" is perhaps more descriptive.  (As opposed to some weighted inner product.)

Missing a close paren after "or scalar product".

Reply:
For me and the professors who have taught me linear algebra, Euclidean inner product implies that the corresponding field are the real numbers. Especially, if a vector space over the complex numbers is considered, the inner product has been called unitary inner product. But if Euclidean inner product means just an inner product in (american) English, I can of course rename it.

Reply:
The close parenthesis is after "or scalar product, i.e. the sum of `inner(x[i],y[i])`)". If it might be better to have this part of the additional explanation outside of the parentheses, I will of course change it.

Reply:
The reason for the qualifier "Euclidean" or similar is to distinguish it from some other weighted inner product.   But since we write that it is the sum of inner(x[i],y[i]) explicitly, I guess we don't need any other qualifiers.

Yes, I would suggest having the `i.e. the sum of ...` part go outside of the parentheses, as this is necessary to precisely define what inner product we are using (as opposed to just a synonym).

Reply:
The docstring should also mention: `dot(x,y)` and `x ⋅ y` (where `⋅` can be typed by tab-completing `\cdot` in the REPL) are synonyms for `inner(x,y)`.